### PR TITLE
CBL-4276, CBL-4277: Remove use of deprecated LiteCore testing API

### PR DESCRIPTION
**Pull request by bmeike**
This is a biggish change.
In addition to simply removing the calls, I've also moved the bindings to a test-only file.  The JNI for the test-only calls will still be in the release product, but the Java bindings for those calls are now part of the test suit and in a separate file that is *not* part of the released product.

### CODE REVIEW

File: common/main/java/com/couchbase/lite/Collection.java [hunk 1]
```diff
@@ -46,7 +46,36 @@
 
 
 /**
+ * A <code>Collection</code> is a container for documents similar to aa table in a relational
```
Comment (bmeike) [inline common/main/java/com/couchbase/lite/Collection.java:49]: This is a backport of the documentation on the Collections class.  Not really related to this PR

File: common/main/java/com/couchbase/lite/internal/core/C4Database.java [hunk 2]
```diff
@@ -93,9 +91,6 @@ void nCopy(
         @NonNull
```
Comment (bmeike) [inline common/main/java/com/couchbase/lite/internal/core/C4Database.java:91]: These are a couple of other functions that are only used by tests.  I took the opportunity to move them to the TestUtils and out of the release code.

File: common/main/java/com/couchbase/lite/internal/support/Log.java [hunk 3]
```diff
@@ -322,6 +323,7 @@ private static void log(
     }
 
     private static boolean shouldLog(@NonNull LogLevel logLevel) {
+        if ((LogLevel.VERBOSE.compareTo(logLevel) >= 0) && !CouchbaseLiteInternal.debugging()) { return false; }
```
Comment (bmeike) [inline common/main/java/com/couchbase/lite/internal/support/Log.java:326]: Another change that is not strictly related to the topic: Don't log below the VERBOSE level, unless debugging.  Debugging *is* configurable at runtime.

File: common/test/java/com/couchbase/lite/LoadTest.kt [hunk 4]
```diff
@@ -38,7 +38,7 @@ class LoadTest : BaseDbTest() {
             "starqlteue" to 150,
             "dandelion_global" to 100,
             "cheetah" to 100,
-            "sunfish" to 55,
+            "sunfish" to 65,
```
Comment (bmeike) [inline common/test/java/com/couchbase/lite/LoadTest.kt:41]: The 2XL device is not as fast as I expected.  Give it a little more time.

File: common/test/java/com/couchbase/lite/LoadTest.kt [hunk 5]
```diff
@@ -209,7 +209,7 @@ class LoadTest : BaseDbTest() {
     @Test
     fun testSaveRevisions2() {
         val mDoc = MutableDocument()
```
Comment (bmeike) [inline common/test/java/com/couchbase/lite/LoadTest.kt:211]: This test has been failing a lot, on multiple devices.  Give it a little more time.

File: common/test/java/com/couchbase/lite/QueryTest.java [hunk 6]
```diff
@@ -1584,7 +1584,6 @@ public void testCountFunctions() {
         verifyQuery(query, 1, (n, result) -> assertEquals(100L, (long) result.getValue(0)));
     }
 
-    @Ignore("Doesn't test anything")
     @Test
```
Comment (bmeike) [inline common/test/java/com/couchbase/lite/QueryTest.java:1587]: This test absolutely *does* test something!!!

File: common/test/java/com/couchbase/lite/internal/core/C4AllDocsPerformanceTest.java [hunk 7]
```diff
@@ -34,8 +34,6 @@
 import static org.junit.Assert.assertNotNull;
 
 
-// !!! MAKE COLLECTION SAVVY
-
 /**
```
Comment (bmeike) [inline common/test/java/com/couchbase/lite/internal/core/C4AllDocsPerformanceTest.java:37]: Moving to the new LiteCore API means these tests had to become collection saavy

File: common/test/java/com/couchbase/lite/internal/core/C4DatabaseTest.kt [hunk 8]
```diff
@@ -284,38 +282,39 @@ class C4DatabaseTest : C4BaseTest() {
             } finally {
                 c4Database.endTransaction(true)
             }
+
             val store = c4Database.blobStore
             assertNotNull(store)
-            compact(c4Database)
+            c4Database.performMaintenance(MaintenanceType.COMPACT)
```
Comment (bmeike) [inline common/test/java/com/couchbase/lite/internal/core/C4DatabaseTest.kt:288]: The "compact" function obfuscated the call that was actually being tested.

**Review by pasin — APPROVED** [review]
